components/controls/FavoriteBase.tsx

```diff
@@ -48,6 +48,12 @@ export default class FavoriteBase extends React.PureComponent<Props, State> {
     this.mounted = true;
   }
 
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.favorite !== this.props.favorite) {
+      this.setState({ favorite: this.props.favorite });
```

Comment:
Good catch, @siegfried-ehret-sonarsource do you remember why this component has a state ?
To me it seems like this component should not have a state and only be controlled by it's parent...

Reply:
I agree. It might looks like this and it seems to solve the problem. At least, it solves my problem and do not break the initial fix (start disappearing when scrolling).

```typescript
export default class FavoriteBase extends React.PureComponent<Props> {
  toggleFavorite = () =>
    this.props.favorite ? this.props.removeFavorite() : this.props.addFavorite();

  render() {
    const { favorite } = this.props;
    const tooltip = favorite
      ? translate('favorite.current', this.props.qualifier)
      : translate('favorite.check', this.props.qualifier);
    const ariaLabel = translate('favorite.action', favorite ? 'remove' : 'add');
    return (
      <Tooltip overlay={tooltip}>
        <ButtonLink
          aria-label={ariaLabel}
          className={classNames('favorite-link', 'link-no-underline', this.props.className)}
          onClick={this.toggleFavorite}>
          <FavoriteIcon favorite={favorite} />
        </ButtonLink>
      </Tooltip>
    );
  }
}

```

Reply:
Yeah it would simplify things so much... I think the initial fix went crazy, unless I don't remember something about this state ?

Reply:
I don't remember :(
If the star doesn't disappear on scroll I am happy with everything.

Reply:
BTW the Favorite component should be in sonar-ui-common no ?

Reply:
> BTW the Favorite component should be in sonar-ui-common no ?

Hmm, I thought so at first, but in fact it is the one responsible for the api call (add/remove). So now I do think it should exists. In fact, I think that it is its only reason to exists (to wrap those calls).

Reply:
Indeed it seems to only exist to wrap those calls, and we don't have access to api methods in sonar-ui-common. So either we drop it or keep it in sonar-web.

Reply:
If I recall correctly, the star state disappeared when scrolling the Projects page. It probably had something to do with the virtualized list we use there. Rings a bell @siegfried-ehret-sonarsource ?